### PR TITLE
Gen3 runners

### PR DESCRIPTION
**Pull request by zoete**
https://github.com/neondatabase/cloud/issues/2738

### CODE REVIEW

**Review by bayandin — COMMENTED** [review]
Is it possible to decouple changing runners and replacing cache actions into two separate PRs?

**Comment by zoete** [comment]
> Is it possible to decouple changing runners and replacing cache actions into two separate PRs?

Done: https://github.com/neondatabase/neon/pull/3304

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -916,8 +932,8 @@ jobs:
           rm -f neon_install.tar.gz .neon_current_version
 
   deploy-prod-new:
-    runs-on: prod
-    container: 093970136003.dkr.ecr.eu-central-1.amazonaws.com/ansible:latest
+    runs-on: [ self-hosted, gen3, small ]
```
[the remark this answers is not in the document]
Comment (zoete) [reply .github/workflows/build_and_test.yml:935]: Good catch, fixing

**Comment by kelvich** [comment]
@zoete can you please write a page on cloud wiki about how access runners and what matrics and log labels that nodes do have (or even better link to the grafana dashboard)

**Pull request merged**